### PR TITLE
`no-single-promise-in-promise-methods`: Remove breaken autofix for `Promise.all()`

### LINKED ISSUE #2385: Problem with autofix for `no-single-promise-in-promise-methods`
Hi,
Thank you for creating and maintaining this plugin.

When autofixing `no-single-promise-in-promise-methods` code might break:

Original code:
```ts
const results = await Promise.all([somePromise]);
doSomething(results[0]);
```

Autofixed code:
```ts
const results = await somePromise;
doSomething(results[0]); // Problem, results is no longer an array
```

I don't know if it's possible to automatically identify the usages of the resolved array and fix them, if not maybe not autofix?

Thanks

### PR DESCRIPTION
**Pull request by Clement398**
Fixes #2385

### CODE REVIEW

File: test/no-single-promise-in-promise-methods.mjs [hunk 1]
```diff
@@ -6,6 +6,8 @@ const {test} = getTester(import.meta);
 // `await`ed
 test.snapshot({
 	valid: [
+		'const results = await Promise.all([promise])',
+		'results = await Promise.all([promise])',
```
Comment (sindresorhus) [inline test/no-single-promise-in-promise-methods.mjs:10]: I don't think this should be valid. The problem mentioned in https://github.com/sindresorhus/eslint-plugin-unicorn/issues/2385 is with the auto fix, not with what the rule reports on.

Comment (Clement398) [reply test/no-single-promise-in-promise-methods.mjs]: It is corrected.

File: test/snapshots/no-single-promise-in-promise-methods.mjs.md [hunk 2]
```diff
@@ -1126,3 +1126,206 @@ Generated by [AVA](https://avajs.dev).
     Suggestion 2/2: Switch to \`Promise.resolve(…)\`.␊
       1 | Promise.resolve(null).then()␊
     `
+
+## invalid(1): Promise.all([promise])
```
Comment (fisker) [inline test/snapshots/no-single-promise-in-promise-methods.mjs.md:1130]: Tests above swithed `Promsie.all` to `Promise.race` to make sure the logic for other methods doesn't change.

**Review by fisker — APPROVED** [review]

**Comment by Clement398** [comment]
Thank you @fisker for all these improvements.

**Pull request merged**